### DISCUSSION TITLE
Registry should reject packages with mismatched name/version

### DISCUSSION THREAD

**Discussion by jmyersmsft** [npm/feedback#797] [Registry]
See for instance node-rdkafka 2.14.0. The package.json inside the tarball reports the version as "v2.14.2-1.HEAD". This causes problems for any system that uses or validates the package.json inside the package. 

I would further argue that the registry should ignore, or reject mismatches in, _any_ field in the uploaded document that ultimately comes from package.json. Allowing mismatches could lead to a security issue if e.g. `scripts` is mismatched. That said, I think name and version are the most important to validate because I think they are the most likely to cause problems in practice because they could effectively lead to putting the package in the wrong place.

EDIT: I've seen #84 but that's more about checking packages relative to other packages, whereas I'm talking about checking both sources of truth for the same upload

**Comment by ljharb** [comment]
It seems like a registry bug to have allowed that package to be published at all - the npm cli should have prevented it's publishing as well.

**Reply by jmyersmsft** [reply]
Yeah, that's what I mean. They must not have used the npm cli to publish because that should have taken the values from the package.json in the first place, but the registry should also have rejected the publish.

**Reply by ljharb** [reply]
Seems worth filing a support ticket about!
